coverage/html.py

```diff
@@ -195,14 +195,18 @@ def html_file(self, fr, analysis):
         source = fr.source()
 
         # Find out if the file on disk is already correct.
-        this_hash = self.file_hash(source.encode('utf-8'), fr)
-        that_hash = self.status.file_hash(rootname)
-        if this_hash == that_hash:
-            # Nothing has changed to require the file to be reported again.
-            self.files.append(self.status.index_info(rootname))
-            return
-
-        self.status.set_file_hash(rootname, this_hash)
+        if os.path.isfile(html_path):
+            this_hash = ""
+            with open(html_path, "rb") as f:
+                hasher = Hasher()
+                hasher.update(f.read().decode("UTF-8"))
+                this_hash = hasher.hexdigest()
+            that_hash = self.status.file_hash(rootname)
```

Comment:
Maybe I'm misunderstanding something. This seems like it will solve your use case (force a file to be re-written if the HTML file has been edited), but it loses the original goal.  This code will now skip writing the HTML file if the current HTML file is the same as when it was last written.  But the source code, or the results, may have changed, requiring the file to be re-written, and this hash won't detect that.

Reply:
There are tests for that behavior, but they are flawed, so they all pass on this branch.

Reply:
I've updated the tests, they are on the nedbat/better-delta-html-tests branch if you want to try them out.

Reply:
But I should warn you: i'm not sure I'm willing to incur a noticeable slow-down to include the hash of the previously generated files.  This seems like an unlikely need, and has a simple work-around.

Reply:
Thanks for the work on this, but it isn't worth the cost. In my experiments, this adds significantly to the cost of HTML generation.

As I understand it, you are modifying all the HTML files. You should delete the directory before running the HTML report.